src/scheme/WeightedLeastSquares.cpp

```diff
@@ -40,55 +40,86 @@ bool WeightedLeastSquares::solve_(const LinearizedControlProblem & problem,
 void WeightedLeastSquares::updateComputationData_(const LinearizedControlProblem & problem,
                                                   internal::ProblemComputationData * data) const
 {
+  using EventType = ProblemDefinitionEvent::Type;
   solver::internal::SolverEvents se;
+  bool fullRebuild = false;
 
   if(data->hasEvents())
   {
     Memory * memory = static_cast<Memory *>(data);
 
-    while(memory->hasEvents())
+    // If some events require to rebuild the data, we skip all other events.
+    for(const auto & e : memory->events())
     {
-      auto e = memory->popEvent();
-      switch(e.type())
+      if(e.type() == EventType::SubstitutionAddition || e.type() == EventType::SubstitutionRemoval)
       {
-        case ProblemDefinitionEvent::Type::WeightChange: {
-          const auto & c = problem.constraintWithRequirements(e.emitter());
-          if(c.requirements->priorityLevel().value() == 0)
-            throw std::runtime_error(
-                "[WeightedLeastSquares::updateComputationData_] "
-                "WeightedLeastSquares does not allow to change the weight of a Task with priority 0.");
-          se.addScalarWeightEvent(c.constraint.get());
-        }
-        break;
-        case ProblemDefinitionEvent::Type::AnisotropicWeightChange: {
-          const auto & c = problem.constraintWithRequirements(e.emitter());
-          if(c.requirements->priorityLevel().value() == 0)
-            throw std::runtime_error(
-                "[WeightedLeastSquares::updateComputationData_] "
-                "WeightedLeastSquares does not allow to change the weight of a Task with priority 0.");
-          se.addVectorWeightEvent(c.constraint.get());
-        }
+        fullRebuild = true;
```

Comment:
Rather than having `fullRebuild`, I suggest:
```
resetComputationData(problem, memory);
return;
```

This will save an indentation level in the remainder and make it easier to understand :)

Reply:
Changed in 8985dab73f16cfa6fc315be162e9b1dd5e18cdfc